atlp-rwanda/commanders-rn-medica
# chore(supabase): Set up Supabase as backend service and add initial tests (#83)

#### What does this PR do?

This PR sets up Supabase as the backend service for our application and includes initial tests for sign-up and sign-in functionality.

#### Description of Task to be completed?

* Initializes the Supabase client with the provided URL and Anon Key.
* Configures authentication settings for AsyncStorage, enabling auto-refresh and session persistence.
* Adds basic tests to verify sign-up and sign-in functionality.

#### How should this be manually tested?

To manually test the changes:

1. Access the sign-up page (if applicable).
2. Enter a valid email address and password.
3. Confirm the password.
4. Click the "Sign Up" button.
5. Verify that the user is redirected to the user profile page (or expected success behavior).
6. Try to sign up with an invalid email address or password.
7. Ensure appropriate error messages are displayed in these scenarios.
8. Repeat steps 1-4 for sign-in functionality.

#### Questions:

* Are there any specific requirements for error handling during sign-up and sign-in?

**Please review these changes and provide feedback.**
